javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -1048,11 +1048,15 @@ public int length() {
 
     @Override
     public Array<T> reverse() {
-        final Object[] arr = new Object[delegate.length];
-        for (int i = 0; i < delegate.length; i++) {
-            arr[delegate.length - 1 - i] = delegate[i];
+        if (size() <= 1) {
+            return this;
+        } else {
+            final Object[] arr = new Object[delegate.length];
+            for (int i = 0; i < delegate.length; i++) {
+                arr[delegate.length - 1 - i] = delegate[i];
+            }
```

Comment:
Indeed, your version makes it obvious that the two iterate in opposite directions, 
We also repeat the `delegate.length - 1`, which is a constant (the JVM might be smart enough to detect it, I don't know).
Your example could be simplified further by inlining `length`: it might also be more optimal for the JVM to know that we go through the whole array (it could unroll the loop better, if it knows the size in advance), and using `i` and `j` in the declaration order:
```java
for (int i = 0, j = delegate.length - 1; i < delegate.length; i++, j--) {
    arr[i] = delegate[j];
}
```